va/va_prot.h

```diff
@@ -0,0 +1,457 @@
+ * using symmetric cryptographic. In DRM world, authentication of client system,
+ * delivery of bitstream and required cryptographic assets to client system is
+ * loosely called provisioning and license acquisition while the processing of
+ * cryptographically secure bitstream is divided as video decryption/decoding,
+ * audio decryption/playback, video display. Besides DRM standards, Video/Audio
+ * bitstream encryption standard such as
+ * [Common Encryption Standard(CENC)](https://www.iso.org/standard/76597.html)
+ * provides a mechanism to normalize bitstream encryption methods across vendors
+ * while providing flexibility.
+ *
+ * \section DRM Pipeline
+ * The most of DRM standards execute following deep pipeline to playback
+ * contents on client systems from streaming serveres - provisioning uses
+ * provisioning servers, licence aquasition uses license server, video
+ * bitstream delivery uses content server and decryption/decoding, audio
+ * bitstream delivery uses content server and decyption/playback,
+ * display/playback.
+ *
+ * \section LibVA Protected Content APIs
+ * The most of DRM standards execute following deep pipeline to playback
+ * contents on client systems from streaming serveres - provisioning uses
+ * provisioning servers, licence aquasition uses license server, video
+ * bitstream delivery uses content server and decryption/decoding, audio
+ * bitstream delivery uses content server and decyption/playback,
+ * display/playback.
+ *
+ * The Provisioning and License Aquasition implementations are Independent
+ * Hardware Vendor (IHV) specific but most IHVs use some form of Trusted
+ * Execution Environment (TEE) to prepare client platform or system for DRM
+ * content playback. The provisioning operations use provisioning server (as
+ * instructed in DRM standard) and client system TEE. The communication between
+ * provisioning server and client system TEE uses asymmetic cryptographic
+ * mechanism. This step provides a way to establish root-of-trust between
+ * client system and streaming servers. Once root-of-trust is established then
+ * client system requests for license aquasition for a particular streaming
+ * title. The license aquasition involves communication between licensing
+ * server and TEE using asymmetic cryptographic mechanism. At end of this step,
+ * client system TEE has required assets to decrypt/decode. Although these
+ * communication does not direcly involve video aspect of GPU but **facilitate
+ * GPU required assets to playback premium contents**.
+ *
+ * To support DRM standard requirements in playback pipeline, OSes and HWs
+ * incorporate various methods to protect full playback pipeline. These methods
+ * of protection could be SW based or HW based. The SW based protection
+ * mechanism of DRMs is called SWDRM while HW based protection mechanism is
+ * called HWDRM. There is no existing support in LibVA to support either DRM
```

Comment:
"There is no existing support in LibVA to support either DRM mechanism."

We can't merge this statement in since the action of merge immediately will invalidate it.

Reply:
This still needs to be fixed.

Reply:
Fixed and updated.

Reply:
Thanks Ryan